cfbs/cfbs_config.py

```diff
@@ -145,27 +147,27 @@ def _convert_added_by(added_by, to_add):
         # Convert list -> dict:
         if not isinstance(added_by, dict):
             assert len(added_by) == len(to_add)
-            added_by = {k: v for k, v in zip(to_add, added_by)}
+            added_by = {k.name: v for k, v in zip(to_add, added_by)}
 
         # Should have a dict with keys for everything in to_add:
-        assert not any((k not in added_by for k in to_add))
+        assert not any((k.name not in added_by for k in to_add))
 
         return added_by
 
     def _update_added_by(self, requested, added_by):
-        for module in self["build"]:
-            if module["name"] in requested:
-                new_added_by = added_by[module["name"]]
-                if new_added_by == "cfbs add":
-                    module["added_by"] = "cfbs add"
+        for req in requested:
+            for mod in self["build"]:
```

Comment:
Would it be possible to do `if req["name"] in self["build"]:`?

Reply:
`self["build"]` is a list (JSON array) so this doesn't work, but we could add a helper, `self.already_added(name)`.